sdks/java/core/src/test/java/org/apache/beam/sdk/testing/PAssertTest.java

```diff
@@ -386,7 +386,7 @@ private static Throwable runExpectingAssertionFailure(Pipeline pipeline) {
     // is first caught by JUnit and causes a test failure.
     try {
       pipeline.run();
-    } catch (AssertionError exc) {
+    } catch (Exception exc) {
```

Comment:
Longer-term, I don't think it has to be the exact inner error, but I do think we need to standardize on the outermost error - users need to be able to switch runners via configuration and have their code (including try/catch of pipeline failures) keep working the same way.